.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,21 +1,46 @@
-#### Why we need this PR?
-*Pick one, remove the rest*
+## Nature of the PR
+
+- [ ] Bugfix
+- [ ] New feature
+- [ ] Refactoring of existing functionality
+- [ ] Documentation update
+- [ ] Other
+
+## Description
+
+*Describe what happened in this PR. For example:*
 - It fixes a bug *bug description*
 - It adds new feature needed because of *why we need this feature*
 - Refactoring to improve *what*
 
-#### What happened in this PR?
-*Fill relevant points, put NA otherwise. Replace anything inside []*
- - What solution was applied:
-     *[ Explain solution of the problem, a new feature added here. ]*
- - Affected modules and functionalities:
-     *[ Describe here what was changed, added, removed. ]*
- - Key points relevant for the review:
-     *[ Describe here what is the most important part that reviewers should focus on. ]*
- - Validation and testing:
-     *[ Describe here if and how this PR is tested. ]*
- - Documentation (including examples):
-     *[ Describe here if documentation and examples were updated. ]*
+*Fill extra details, if applicable*
+- What solution was applied:
+    *[ Explain solution of the problem, a new feature added here. ]*
+- Affected modules and functionalities:
+    *[ Describe here what was changed, added, removed. ]*
+- Key points relevant for the review:
+    *[ Describe here what is the most important part that reviewers should focus on. ]*
+
+## Checklist
+
+### Requirements
+
+- [ ] Implements new requirements
+- [ ] Affects existing requirements
+- [ ] N/A
+
+**REQ IDs**: *[ Introduce new or affected requirement IDs, if applicable ]*
+
+### Tests
+
+- [ ] Existing tests apply
+- [ ] New tests added
+- [ ] N/A
+
+### Documentation
 
+- [ ] Existing documentation apply
```

Comment:
```suggestion
- [ ] Existing documentation applies
```
However, it would be nice to have some more boxes (Docstrings, Doxygen, RST, Ohter).